[[ Bug 14554 ]] Add user guides for widgets

livecodeali opened the pull request.

livecodeali committed:
[[ Spinner ]] Add user guide

livecodeali committed:
[[ GraphWidget ]] Add user guide

livecodeali committed:
[[ SVGPath ]] Add user guide

livecodeali committed:
[[ HeaderWidget ]] Add user guide

livecodeali committed:
[[ NavbarWidget ]] Add user guide

livecodeali committed:
[[ SegmentedWidget ]] Add user guide

livecodeali committed:
[[ Browser ]] Add user guide

livecodeali committed:
[[ Clock ]] Add user guide

livecodeali committed:
[[ SwitchButton ]] Add user guide

livecodeali committed:
[[ TreeView ]] Add user guide

montegoulding commented on extensions/widgets/browser/docs/guide/guide.md:
```diff
@@ -0,0 +1,55 @@
+
+The most straightforward way to use the browser widget is to set its 
+`url` property to display a web page. It can also be set to a local file 
+url, for example to display a PDF.
```
PDF is probably not the best suggestion as its not supported on all platforms

montegoulding commented on extensions/widgets/browser/docs/guide/guide.md:
```diff
@@ -0,0 +1,55 @@
+
+## Controlling navigation
+The browser sends the following messages which can be used to control navigation within the browser. See the API documentation for more details:
+* browserDocumentLoadBegin
```
add a line above the list here

montegoulding commented on extensions/widgets/clock/docs/guide/guide.md:
```diff
@@ -0,0 +1,29 @@
+![Clock widget](images/clock.png)
+
+## Creating a Clock
+A Clock widget can be created by dragging it out from the Tools
```
We probably need to decide if we want to capitalise the widget name throughout. I think this is an issue for most of the guides.

livecodeali replied:
Good point... It looks like in most of the other guides, title case for section headings is standard. So I think this is one of the few correct cases.

montegoulding commented on extensions/widgets/graph/docs/guide/guide.md:
```diff
@@ -0,0 +1,91 @@
+
+![Line graph with no grid or markers](images/data-no-lines.png)
+	
+This is helpful for creating a scatter plot.
```
Due to the treatment of x axis as labels we can't use the widget as a scatter plot at the moment so perhaps delete this line?

montegoulding commented on extensions/widgets/graph/docs/guide/guide.md:
```diff
@@ -0,0 +1,91 @@
+	put "Jul,18" into tHilitedDataPoint
+	set the hilitedCoordinates of widget "Line Graph" to tHilitedDataPoint
+
+![Line graph with hilited coordinates](images/data-hilited-coordinates.png)
```
Ohh... optimal month or @montegoulding to visit Scotland ;-)

livecodeali committed:
[[ WidgetGuides ]] Various tweaks
Ensure capitalisation is consistent, i.e. title case all section
headings and don't capitalise widget names elsewhere.
Also fix a few misleading sentences.

montegoulding commented:
@livecode-vulcan review ok b337118

livecode-vulcan commented:
💙 review by @montegoulding ok b33711826d94257360e71cf31850964d558b7894

livecode-vulcan commented:
😎 test success b33711826d94257360e71cf31850964d558b7894

* try-community-armv6-android-api8: [success](https://vulcan.livecode.com/builders/try-community-armv6-android-api8/builds/1379)
* try-community-armv6-android-api9: [success](https://vulcan.livecode.com/builders/try-community-armv6-android-api9/builds/1395)
* try-community-js-emscripten-sdk1.35: [success](https://vulcan.livecode.com/builders/try-community-js-emscripten-sdk1.35/builds/1369)
* try-community-universal-ios-iphoneos11.2: [success](https://vulcan.livecode.com/builders/try-community-universal-ios-iphoneos11.2/builds/322)
* try-community-universal-ios-iphonesimulator11.2: [success](https://vulcan.livecode.com/builders/try-community-universal-ios-iphonesimulator11.2/builds/320)
* try-community-universal-mac-macosx10.6: [success](https://vulcan.livecode.com/builders/try-community-universal-mac-macosx10.6/builds/1373)
* try-community-universal-mac-macosx10.9: [success](https://vulcan.livecode.com/builders/try-community-universal-mac-macosx10.9/builds/1399)
* try-community-x86-linux-debian7: [success](https://vulcan.livecode.com/builders/try-community-x86-linux-debian7/builds/1379)
* try-community-x86-linux-debian8: [success](https://vulcan.livecode.com/builders/try-community-x86-linux-debian8/builds/1393)
* try-community-x86_64-linux-debian7: [success](https://vulcan.livecode.com/builders/try-community-x86_64-linux-debian7/builds/1378)
* try-community-x86_64-linux-debian8: [success](https://vulcan.livecode.com/builders/try-community-x86_64-linux-debian8/builds/1388)
* try-community-x86-win32: [success](https://vulcan.livecode.com/builders/try-community-x86-win32/builds/1429)
* try-community-x86_64-win32: [success](https://vulcan.livecode.com/builders/try-community-x86_64-win32/builds/1369)
